ics_space_amplification_test.py

```diff
@@ -0,0 +1,218 @@
+            written_data_size_gb=total_new_data_to_write_gb, start_time=start_time)
+        verify_nodes_space_amplification(dict_nodes_space_amplification=dict_nodes_space_amplification)
+
+        self.log.debug('Test Space-amplification on over-write data')
+        prepare_overwrite_cmd = "cassandra-stress write cl=ALL  n={overwrite_ops_num} -schema 'replication(factor=3) compaction(strategy=LeveledCompactionStrategy)' -port jmx=6868 -mode cql3 native" \
+                                " -rate threads=1000 -col 'size=FIXED({column_size}) n=FIXED(num_of_columns)' -pop 'dist=uniform(1..{overwrite_ops_num})' ".format(
+                                    **locals())
+
+        verify_overwrite_queue = list()
+        self.log.debug('Total data to write per cycle is: {} GB '.format(total_data_to_overwrite_gb))
+
+        self.wait_no_compactions_running()
+        overwrite_cycles_num = 4
+        for i in range(1, overwrite_cycles_num + 1):
+            self.log.debug('Starting overwrite stress cycle {}..'.format(i))
+            dict_nodes_capacity_before_overwrite_data = self._get_nodes_used_capacity()
+            start_time = time.time()
+            self._run_all_stress_cmds(verify_overwrite_queue, params={'stress_cmd': prepare_overwrite_cmd,
+                                                                      'keyspace_num': keyspace_num})
+            for stress in verify_overwrite_queue:
+                self.verify_stress_thread(cs_thread_pool=stress)
+
+            dict_nodes_space_amplification = self._get_nodes_space_amplification_after_write(
+                dict_nodes_initial_capacity=dict_nodes_capacity_before_overwrite_data,
+                written_data_size_gb=total_data_to_overwrite_gb, start_time=start_time)
+            verify_nodes_space_amplification(dict_nodes_space_amplification=dict_nodes_space_amplification)
+
+        self.log.debug('Test Space-amplification on major compaction')
+        dict_nodes_capacity_before_major_compaction = self._get_nodes_used_capacity()
+        start_time = time.time()
+        for node in self.db_cluster.nodes:
+            node.run_nodetool("compact")
+        self.wait_no_compactions_running()
+        dict_nodes_space_amplification = self._get_nodes_space_amplification(
+            dict_nodes_initial_capacity=dict_nodes_capacity_before_major_compaction,
+            start_time=start_time)
+        verify_nodes_space_amplification(dict_nodes_space_amplification=dict_nodes_space_amplification)
+
+
+def verify_nodes_space_amplification(dict_nodes_space_amplification):
+    for node_ip, space_amplification_gb in dict_nodes_space_amplification.items():
+        assert space_amplification_gb < MAX_ICS_SPACE_AMPLIFICATION_ALLOWED_GB, \
+            f'Node {node_ip} space amplification of: {space_amplification_gb} exceeds the maximum allowed ({MAX_ICS_SPACE_AMPLIFICATION_ALLOWED_GB})'
+
+
+def generate_capacity_query_postfix(node):
```

Comment:
this sound more like `limit_query_to_specific_node()`

Reply:
renamed to 'generate_node_capacity_query_postfix'